tests/unit/test_bruteforce_multi.cpp

```diff
@@ -710,6 +710,50 @@ TYPED_TEST(BruteForceMultiTest, search_empty_index) {
     VecSimIndex_Free(index);
 }
 
+TYPED_TEST(BruteForceMultiTest, removeVectorWithSwaps) {
+    size_t dim = 4;
+    size_t n = 6;
+
+    BFParams params = {.dim = dim, .metric = VecSimMetric_L2};
+    auto *index = this->CastToBF_Multi(this->CreateNewIndex(params));
+
+    // Insert 3 vectors under two different labels, so that we will have:
+    // {first_label->[0,1,3], second_label->[2,4,5]}
+    labelType first_label = 1;
+    labelType second_label = 2;
+
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, first_label);
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, first_label);
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, second_label);
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, first_label);
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, second_label);
+    GenerateAndAddVector<TEST_DATA_T>(index, dim, second_label);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+
+    // Artificially reorder the internal ids to test that we make the right changes
+    // when we have an id that appears twice in the array upon deleting the ids one by one.
+    ASSERT_EQ(index->labelToIdsLookup.at(second_label).size(), n / 2);
+    index->labelToIdsLookup.at(second_label)[0] = 4;
+    index->labelToIdsLookup.at(second_label)[1] = 2;
+    index->labelToIdsLookup.at(second_label)[2] = 5;
```

Comment:
consider adding assertions to "prove" that the ids of the second label are 2, 4, and 5 before setting them to these values